src/context/yaml/handlers/actions.ts

```diff
@@ -86,7 +86,8 @@ async function dump(context: YAMLContext): Promise<ParsedActions> {
       runtime: action.runtime,
       dependencies: action.dependencies || [],
       status: action.status,
-      secrets: mapSecrets(action.secrets || []),
+      secrets:
+        typeof action.secrets === 'string' ? action.secrets : mapSecrets(action.secrets || []), //Enables keyword preservation to operate on action secrets
```

Comment:
Most array preservation will work as-is for all YAML handlers, but there is this one instance where a string value for `action.secrets` would trigger a runtime error when applying a map function.